usersession/agent/rest_api.go

```diff
@@ -349,3 +355,50 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(nil)
 }
+
+func postRebootRequiredNotification(c *Command, r *http.Request) Response {
+	if ok, resp := validateJSONRequest(r); !ok {
+		return resp
+	}
+
+	decoder := json.NewDecoder(r.Body)
+
+	var rebootRequired client.RebootRequiredInfo
+	if err := decoder.Decode(&rebootRequired); err != nil {
+		return BadRequest("cannot decode request body into reboot required notification info: %v", err)
+	}
+
+	// Note that since the connection is shared, we are not closing it.
+	if c.s.bus == nil {
+		return SyncResponse(&resp{
+			Type:   ResponseTypeError,
+			Status: 500,
+			Result: &errorResult{
+				Message: "cannot connect to the session bus",
+			},
+		})
+	}
+
+	summary := fmt.Sprintf(i18n.G("Update available for %q."), rebootRequired.InstanceName)
+	body := i18n.G("Restart the system to update now.")
```

Comment:
I know you've said the UI needs design input, but which of these two is the more important piece of information? The fact that a reboot is required, or that a particular snap has an update available? Would it make sense to switch these over?

Reply:
I ran the text past design as is. I would think the update is the more important information - we want to encourage the user to reboot but they don't have to.